src/textlint-engine.js

```diff
@@ -1,217 +1,60 @@
-            return this.textLint.fixFile(file);
-        });
-        return Promise.all(results);
-    }
-
-    /**
-     * Fix texts with ext option.
-     *
-     * @param {string} text linting text content
-     * @param {string} ext ext is a type for linting. default: ".txt"
-     * @returns {TextLintFixResult[]}
-     */
-    fixText(text, ext = ".txt") {
-        // filepath or ext
-        const actualExt = ext[0] === "." ? ext : path.extname(ext);
-        if (actualExt.length === 0) {
-            throw new Error("should specify the extension.\nex) .md");
-        }
-        return this.textLint.fixText(text, actualExt).then(result => {
-            return [result];
-        });
-    }
-
-    /**
-     * format {@link results} and return output text.
-     * @param {TextLintResult[]} results the collection of result
-     * @returns {string} formatted output text
-     * @example
-     *  console.log(formatResults(results));
-     */
-    formatResults(results) {
-        const formatterConfig = {
-            formatterName: this.config.formatterName,
-            color: this.config.color
+     * TextLintEngine is a adaptor of TextLintEngineCore.
+     * @param {Config} config
+     * @returns {TextLintEngineCore}
+     */
+    constructor(config) {
+        const executor = {
+            /**
+             * @param {TextLintCore} textlintCore
+             * @returns {function()}
+             */
+            onFile: (textlintCore) => {
+                /**
+                 * Executes the current configuration on an array of file and directory names.
+                 * TextLintEngine#executeOnFile
+                 * @param {String[]}  files An array of file and directory names.
+                 * @returns {TextLintResult[]} The results for all files that were linted.
+                 */
+                return function executeOnFile(file) {
+                    return textlintCore.lintFile(file);
+                };
+            },
+            /**
+             * @param {TextLintCore} textlintCore
+             * @returns {function()}
+             */
+            onText: (textlintCore) => {
+                /**
+                 * lint text, and return TextLintResult[]
+                 * TextLintEngine#executeOnText
+                 * @param {string} text linting text content
+                 * @param {string} ext ext is a type for linting. default: ".txt"
+                 * @returns {TextLintResult[]}
+                 */
+                return function executeOnText(text, ext) {
+                    return textlintCore.lintText(text, ext);
+                };
+            },
+            /**
+             * @param {TextLintFormatterOption} formatterConfig
+             */
+            onFormat: (formatterConfig) => {
+                // default formatter name: stylish
+                if (!formatterConfig.formatterName) {
+                    formatterConfig.formatterName = "stylish";
```

Comment:
:memo: XXXengine has default options.
It is better to note the fact in `Config`